device_backends/include/AsyncAccessorManager.h

```diff
@@ -45,13 +32,17 @@ namespace ChimeraTK {
     virtual unsigned int getNumberOfSamples() = 0;
     virtual const std::string& getUnit() = 0;
     virtual const std::string& getDescription() = 0;
-    virtual bool isWriteable() = 0;
+
+    /** Fill the send buffer with data and version number. It is implementation specific where this information is
+     * coming from.
+     */
+    virtual void fillSendBuffer() = 0;
   };
 
   /** Helper class to have a complete descriton to create an Accessor.
    *  It contains all the information given to DeviceBackend:getNDRegisterAccessor, incl. the offset in the
-   *  register which is not known to the accessor itself. Just just to keep the number of parameters for
-   * createAsyncVariable in check.
+   *  register which is not known to the catalogue entry and the accessor itself. Just just to keep the number of
+   * parameters for createAsyncVariable in check.
    */
   struct AccessorInstanceDescriptor {
```

Comment:
Although the constructor body is empty, it is not generated by default. It has all struct variables as argument, so it can be created and filled in one command. Otherwise one would have to write 6 lines instead of one (one to create the struct and one to set each of the parameters).

Reply:
This code works:

```
#include <iostream>
#include <string>

struct Test {
  int a;
  float b;
  std::string c;
};


int main() {
  Test x{42, 123.456, "Hallo Welt"};
  std::cout << x.a << " " << x.b << " " << x.c << std::endl;
}
```

What's the difference here?

Reply:
Ah, C++ brace initialiser. I was not aware that you can use it like that on all the members. I will remove the constructor then.